src/restd.c

```diff
@@ -22,33 +22,36 @@
  * in order to interpret the same data values as were represented by
  * the input local Table D descriptor.
  *
- * @param lun - Internal Fortran I/O stream index associated with BUFR
+ * @param lun - File ID.
  * file
  * @param tddesc - Bit-wise representation of FXY value for local Table
- * D descriptor
+ * D descriptor.
  * @param nctddesc - Number of WMO-standard child descriptors returned
- * in ctddesc
+ * in ctddesc.
  * @param ctddesc - Array of WMO-standard child descriptors equivalent
- * to tddesc
+ * to tddesc.
  *
  * @author J. Ator @date 2004-08-18
 */
 
 void
 restd(f77int *lun, f77int *tddesc, f77int *nctddesc, f77int *ctddesc)
 {
+    static const int ADN_STR_LEN = 7;
+    static const int NEMO_STR_LEN = 9;
```

Comment:
I'm very glad these are now deffined, because we should not have naked constants in our code.

However, in C these would typically be defined with the preprocessor. The reason is that many such constants will end up migrating up to a .h file, when you find that you are using the same constant in multiple places.

This has the minor advantage of not putting them on the stack at all, since the pre-processor is run before compilation.

So instead of defining static ints, use:

```
#define ADN_STR_LEN 7
#define NEMO_STR_LEN 9

```

Did you include the NULL char in the string len? If so, the convention is not to. In other words:

```
#define ADN_STR_LEN 6
char adn_str[ADN_STR_LEN + 1];

```
In other words, make the STR_LEN equal to the length of the string without the NULL, and make sure you add the extra +1 whenever you allocate storage. It might seem funny, but this convention makes everything work well with the C string library, which uses the same style.

Reply:
I did include NULL in these string lengths, and I could redefine everything with the +1 to make it clearer, but FYI that the allocated size also needs to be passed as an argument to the _f routines that are called by this routine.  It's needed for use within copy_f_c_str which is called inside of those _f wrapper routines, for copying between Fortran and C strings to make sure we don't overflow the allocated C space.  So this means I'd also need to include the +1 in those arguments, and not just when initially allocating the storage.